src/main/java/io/cryostat/net/web/http/api/v2/graph/StartRecordingOnTargetMutator.java

```diff
@@ -137,6 +137,11 @@ public HyperlinkedSerializableRecordingDescriptor getAuthenticated(
                     if (settings.containsKey("maxSize")) {
                         builder = builder.maxSize((Long) settings.get("maxSize"));
                     }
+                    boolean archiveOnStop = false;
+                    if (settings.containsKey("archiveOnStop")) {
+                        Boolean v = (Boolean) settings.get("archiveOnStop");
+                        archiveOnStop = v != null && v;
```

Comment:
I think here we don't have to check null? Or is there special use here?

Reply:
Just being defensive. Since the `settings` is a `Map<String, Object>`, the `.get()` is a boxed `Boolean`. If it's a `null` value then unboxing that to the primitive `boolean` would NPE.

Reply:
Oh oh right! That makes sense! Thanks :D